docs: add security gates to fork PR mirroring (#19438)
DataDog/dd-trace-py: docs/contributing.rst

tied to a pull request owned by this repository, and GitLab never mirrors branches from
external forks.

If you're a maintainer merging a fork PR, the mirror moves its code into a trusted CI context with
access to credentials. The reviewer owns that trust decision. Follow this process:

#. Complete the normal code review on the original fork PR. Before mirroring, you should be willing
   to merge its exact current head commit as long as the final CI checks pass.
#. Review everything the internal CI pipeline might execute with extra scrutiny. This includes
   workflow definitions, scripts, tests, build and install configuration, dependency changes,
   network calls, artifact uploads, and code that accesses credentials or changes permissions.
   Involve the relevant code owners or security reviewers when appropriate.
#. Immediately before mirroring, verify that the original PR's head SHA is still the SHA you
   reviewed. If it changed, review the updates and repeat this check.
#. Push that exact reviewed head commit to a new branch in this repository, e.g.
   ``git push origin <fork-pr-head-sha>:refs/heads/<you>/mirror-<PR#>``.

> #. Open a draft shadow PR from that branch into ``main`` with a title that begins

Author: ```suggestion
#. Open a draft shadow PR from that branch into ``main`` with a title that ends with
```

>    ``[DO NOT MERGE]``. This gives the commit a repository-owned PR context, which is required for

Author: ```suggestion
   ``[DO NOT MERGE]`` and follows conventional commits. This gives the commit a repository-owned PR context, which is required for
```

   CI to authenticate correctly while making the shadow PR's purpose clear.
#. Wait for the shadow PR's checks to pass. The identical SHA establishes source-commit parity;
   it does not mean both PRs run in the same context or establish that the code is safe. That
   assurance comes from the review before mirroring.
#. Comment ``/merge`` on the *original* fork PR once its checks are green.
#. After the original fork PR lands, close the shadow PR without merging it, and delete its branch.

> Never mirror speculative, partially reviewed, or unreviewed code just to obtain CI results.

Author: ```suggestion
Never mirror speculative, partially reviewed, or code that has not been reviewed just to obtain CI results.
```
